core/src/test/java/org/bptlab/cepta/count-trains-in-station-tests/CountTrainsInStationTests.java

```diff
@@ -0,0 +1,125 @@
+package org.bptlab.cepta;
+
+import com.google.protobuf.util.Timestamps;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.bptlab.cepta.models.events.correlatedEvents.CountOfTrainsAtStationEventOuterClass.*;
+import org.bptlab.cepta.models.events.train.LiveTrainDataOuterClass;
+import org.bptlab.cepta.operators.CountOfTrainsAtStationFunction;
+import org.bptlab.cepta.providers.LiveTrainDataProvider;
+import org.bptlab.cepta.utils.functions.StreamUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+
+public class CountTrainsInStationTests {
+
+    @Test
+    public void TestCorrectAmountOfTriggers() throws IOException, ParseException {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        int commonStation = 1;
+        LiveTrainDataOuterClass.LiveTrainData first =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(1)
+                        .setEventTime(Timestamps.parse("1972-01-01T10:00:20.021-05:00")).build();
+        LiveTrainDataOuterClass.LiveTrainData second =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(2)
+                        .setEventTime(Timestamps.parse("1972-01-01T10:00:21.022-05:00")).build();
+        LiveTrainDataOuterClass.LiveTrainData end =
+                LiveTrainDataProvider.getDefaultLiveTrainDataEvent().toBuilder()
+                        .setStationId(commonStation)
+                        .setTrainId(3)
+                        .setEventTime(Timestamps.parse("2042-01-01T10:00:31.023-05:00")).build();
```

Comment:
Didn't you plan to make convenient functions for this stuff as part of the test utils stuff? This looks kinda convoluted, so please make a helper function in the provider to avoid this builder stuff.

Reply:
Do you have a good idea on how to make a good helper function? We could create a helper function for **each** group of fields we need to set but that also seems whack.
I agree that this would be better to have smaller creation functions, but i guess we now see what data we are testing on